fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/api/CreditBureauIntegrationAPI.java

```diff
@@ -0,0 +1,160 @@
+        this.creditReportReadPlatformService = creditReportReadPlatformService;
+        this.toApiJsonSerializer = toApiJsonSerializer;
+
+    }
+
+    @POST
+    @Path("creditReport")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String postCreditReport(@Context final UriInfo uriInfo, @RequestParam("params") final Map<String, Object> params) {
+
+        Gson gson = new Gson();
+        final String json = gson.toJson(params);
+        final CommandWrapper commandRequest = new CommandWrapperBuilder().getCreditReport().withJson(json).build();
+
+        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+        return this.toCreditReportApiJsonSerializer.serialize(result);
+
+    }
+
+    // saves fetched-creditreport into database
+    @POST
+    @Path("saveCreditReport")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String saveCreditReport(@Parameter(hidden = true) final String apiRequestBodyAsJson,
+            @QueryParam("creditBureauId") @Parameter(description = "creditBureauId") final Long creditBureauId,
+            @QueryParam("creditReportNumber") @Parameter(description = "creditReportNumber") final String creditReportNumber) {
+
+        final CommandWrapper commandRequest = new CommandWrapperBuilder() //
+                .saveCreditReport(creditBureauId, creditReportNumber) // creditReportNumber is a NRC number for
+                                                                      // Thitsawork
+                .withJson(apiRequestBodyAsJson) // apiRequestBodyAsJson is a creditReport
+                .build(); //
+
+        final CommandProcessingResult result = this.commandsSourceWritePlatformService.logCommandSource(commandRequest);
+        return this.toCreditReportApiJsonSerializer.serialize(result);
+
+    }
+
+    // fetch saved creditReports(NRC) from DB by creditBureauId, to select for downloading and deleting the reports
+    @GET
+    @Path("creditReport/{creditBureauId}")
+    @Consumes({ MediaType.APPLICATION_JSON })
+    @Produces({ MediaType.APPLICATION_JSON })
+    public String getSavedCreditReport(@PathParam("creditBureauId") @Parameter(description = "creditBureauId") final Long creditBureauId,
+            @Context final UriInfo uriInfo) {
+
+        this.context.authenticatedUser();
+
+        final Collection<CreditReportData> creditReport = this.creditReportReadPlatformService.retrieveCreditReport(creditBureauId);
+
+        final ApiRequestJsonSerializationSettings settings = this.apiRequestParameterHelper.process(uriInfo.getQueryParameters());
+        return this.toApiJsonSerializer.serialize(settings, creditReport, RESPONSE_DATA_PARAMETERS);
+
+    }
+
+    // deletes saved creditReports from database
+    @DELETE
+    @Path("deleteCreditReport/{creditBureauId}")
```

Comment:
rename creditBureauID to creditReportID

Reply:
we will need two parameters(creditBureauId/nationalId) to delete the saved CreditReports in database. So creditBureauId is passing from pathparam and nationalId is passing from body

Reply:
alright